File: apply-migration.js
#!/usr/bin/env node

/**
 * Apply Supabase database migration
 * This script executes the SQL migration directly via Supabase's pgAdmin interface
 */

const fs = require('fs');
const https = require('https');

const SUPABASE_URL = 'https://ecksfrhzvfzmfdhhnses.supabase.co';
const SERVICE_ROLE_KEY = 'seyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImVja3Nmcmh6dmZ6bWZkaGhuc2VzIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3OTg5MDQ3NCwiZXhwIjoyMDk1NDY2NDc0fQ.8ocVluF67uFQrIZIP4MU1UyPVXxR1d4oKT4Z7os3OKQ';

// Read the migration SQL
const migrationSQL = fs.readFileSync('./database/02-add-missing-fields.sql', 'utf8');

console.log('🚀 Applying Supabase database migration...\n');
console.log('📝 Migration SQL:');
console.log('----------------------------------------');
console.log(migrationSQL.split('\n').slice(0, 20).join('\n'));
console.log('... (truncated)\n');

// Since we can't directly execute SQL via REST API, we need to use the SQL editor
// This script shows the commands to run

console.log('⚠️  MANUAL STEP REQUIRED\n');
console.log('Unfortunately, Supabase SQL execution requires the web UI for authentication.');
console.log('Please follow these steps:\n');

console.log('1. Open: https://supabase.com/dashboard/project/ecksfrhzvfzmfdhhnses/sql/1');
console.log('2. Click "New Query" button');
console.log('3. Copy and paste the SQL below:');
console.log('----------------------------------------');
console.log(migrationSQL);
console.log('----------------------------------------');
console.log('4. Click "Run" button (blue button on right)');
console.log('5. Wait for "Success" message\n');

console.log('After migration is applied:');
console.log('- npm run build');
console.log('- npm run start');
console.log('- Test daily login rewards\n');

console.log('Need help? See: SUPABASE_MIGRATION_GUIDE.md\n');

File: apply_migration.py
#!/usr/bin/env python3
"""
Apply Supabase database migration using direct PostgreSQL connection
"""

import os
import sys

try:
    import pg8000.native
except ImportError:
    print("❌ pg8000 not available. Installing...")
    os.system("pip install pg8000 --break-system-packages -q")
    import pg8000.native

def load_env():
    """Load environment variables from .env.local"""
    env_file = '.env.local'
    if not os.path.exists(env_file):
        print(f"❌ {env_file} not found")
        sys.exit(1)

    env = {}
    with open(env_file) as f:
        for line in f:
            line = line.strip()
            if line and '=' in line and not line.startswith('#'):
                key, val = line.split('=', 1)
                env[key.strip()] = val.strip()
    return env

def get_connection_string(supabase_url):
    """Convert Supabase URL to PostgreSQL connection string"""
    # Extract project ID from URL: https://[project-id].supabase.co
    project_id = supabase_url.replace('https://', '').replace('.supabase.co', '')

    # Note: We need the database password which is not in .env.local
    # Supabase uses: postgresql://postgres:[password]@[project-id].db.supabase.co:5432/postgres

    return f"postgresql://postgres@{project_id}.db.supabase.co:5432/postgres"

def apply_migration():
    """Apply the database migration"""
    print("🚀 Supabase Database Migration Tool\n")

    env = load_env()
    supabase_url = env.get('NEXT_PUBLIC_SUPABASE_URL')
    service_key = env.get('SUPABASE_SERVICE_ROLE_KEY')

    if not supabase_url or not service_key:
        print("❌ Missing Supabase credentials")
        sys.exit(1)

    print(f"📍 Supabase Project: {supabase_url}\n")

    # Read migration SQL
    if not os.path.exists('database/02-add-missing-fields.sql'):
        print("❌ Migration file not found: database/02-add-missing-fields.sql")
        sys.exit(1)

    with open('database/02-add-missing-fields.sql') as f:
        migration_sql = f.read()

    print("📝 Migration SQL Preview:")
    print("----------------------------------------")
    for line in migration_sql.split('\n')[:15]:
        print(line)
    print("... (truncated)\n")

    print("⚠️  ISSUE: Database Password Required\n")
    print("To apply this migration, we need the PostgreSQL password for:")
    print(f"  {get_connection_string(supabase_url)}\n")

    print("📖 Alternative Methods:\n")

    print("Option 1: Via Supabase Dashboard (Recommended)")
    project_id = supabase_url.split('.')[0].replace('https://', '')
    print(f"  1. Go to: https://supabase.com/dashboard/project/{project_id}/sql/1")
    print("  2. Click 'New Query'")
    print("  3. Paste SQL from: database/02-add-missing-fields.sql")
    print("  4. Click 'Run'\n")

    print("Option 2: Via psql Command Line")
    print(f"  psql 'postgresql://postgres:[password]@{project_id}.db.supabase.co:5432/postgres' < database/02-add-missing-fields.sql\n")

    print("Option 3: Via Supabase CLI")
    print("  supabase db push --linked\n")

    print("📚 Full Documentation: SUPABASE_MIGRATION_GUIDE.md")

if __name__ == '__main__':
    apply_migration()
